app/javascript/src/component_expander.js

```diff
@@ -1,111 +1,160 @@
 
-
-/* Main class
- * Select an element that should show/hide its content based on activator toggle.
- * The component will inject a container that wraps the elements content.
- * An activator will be created, if not specified, and moved to be placed just
- * before the injected container element. Toggle of the activator will show/hide
- * the content, which is now wrapped by the injected container.
- *
- * @$node  (jQuery node) Element found in template that should be enhanced.
- * @config (Object) Configurable key/value pairs.
- **/
+/**
+ * @param $node (jQuery) Html Element to be enhanced.
+ * @param condif (Object) Key/value pairs for config (see above)
+**/
 class Expander {
+  #config;
+  #state;
+
   constructor($node, config) {
-    var conf = utilities.mergeObjects({
-      activator_text: "Toggle", // Text for any self-created activator button.
-      auto_open: false, // Set whether open on creation.
-      duration: 0, // Number determining how long the animation will run
-      $activator: null // Pass in the jQuery element you want to toggle open/close.
+    const conf = mergeObjects({
+      activator: null,
+      wrap_content: true,
+      auto_open: false,
+      duration: 0,
     }, config);
 
-    var id = utilities.uniqueString("Expander_");
-    var $children = $node.children();
-    var $container = $("<div></div>");
+    if(!conf.activator) {
+      console.warn('Cannot initialise an Expander without an activator element.');
+      return;
+    }
 
-    $container.attr("id", id);
-    $container.addClass("Expander_container");
     $node.addClass("Expander");
     $node.data("instance", this);
 
-    this._config = conf;
+    this.#config = conf;
+    this.#state = 'open';
     this.$node = $node;
-    this.$container = $container;
-    this.$activator = Expander.createActivator.call(this, $container);
-
-    // A little accessibility
-    this.$activator.attr("aria-controls", id);
-
-    // Setup and initial state
-    this.$container.append($children);
-    this.$node.prepend(this.$container);
-    this.$node.prepend(this.$activator);
 
-    if(conf.auto_open) {
-      this.open();
-    }
-    else {
-      this.close();
+    const id = uniqueString("Expander_");
+
+    if(typeof conf.activator == 'string') {
+      // We create a button using the title for a label and prepend it to the 
+      // container to toggle disclosure
+      this.$button = this.#createButton(id, conf.activator);
+      $node.prepend(this.$button);
+    } else if (conf.activator.is('button')) {
+      // we enhance the button with the required aria attributes and event
+      // listener 
+      this.$button = this.#enhanceButton(conf.activator, id);
+    } else {
+      // We enhance the title element by wrapping its contents with a button
+      let $activator = conf.activator;
+      $activator.wrapInner( this.#createButton(id) );
+      $activator.addClass("Expander__title");
+      this.$button = $activator.find('button');
```

Comment:
Not keen on multiple `this.$whatever = blah` declarations throughout the constructor. 
I'd say it would be a lot cleaner to use an internal variable in the _thing_ setup, then keep to the one 
`this.public_thing = internal_thing` at the end of the constructor, like (most/all) other components do.

e.g.
```
constructor() {
    var $internal_whatever;
    ...stuff...

    if something
       $internal_whatever = one_thing
    ...stuff...

    if something_else
      $internal_whatever = another_thing
    ... stuff ...

    // Just done once near the end of constructor, like other component code.  
    this.$public_whatever = $internal_whatever
}
```